### ISSUE TITLE
Cancelling the stream with status Status{code=INTERNAL, description=Completed without a response, cause=null}

### ISSUE THREAD

**Issue by AviAlche** [grpc/grpc-java#6568]
We have a gRPC client and server that both are streaming to each other. Code implemented in java.

The client triggers onNext in the server a few times and the server aggregates data each time, the the client then triggers onComplete which starts some server side logic and when it finished the server triggers the onComplete on the client side.

I can see in the logs the following logic:
Server:
Nov 24, 2019 9:19:17 PM io.grpc.internal.ServerCallImpl internalCloseWARNING: Cancelling the stream with status Status{code=INTERNAL, description=Completed without a response, cause=null}

Then a second later the client:
ERROR [grpc-default-executor-42] .... CANCELLED: HTTP/2 error code: CANCEL .. 
Received Rst Stream.

**Comment by ran-su** [comment]
It looks like your server is calling `onComplete` be for sending any response. For any of the streams send at least once message is required before peaceful complete.

**Comment by AviAlche** [comment]
Ive got the following lines in the server onComplete one exactly after the other:
                    responseObserver.onNext(response);
                    responseObserver.onCompleted();

Not sure how long it will take the client::onNext to complete, but is it possible that the grpc does not understand that there is a current message processing in the onNext and therefor cancelling the stream with a "Completed without a response" error.
So basically what im asking is onNext and onComplete block each other?

**Comment by ran-su** [comment]
Your use case sounds more like [this](https://github.com/grpc/grpc-java/blob/6b04fc18f30deb3519c389fe457b99cf5a8d99d0/examples/src/main/java/io/grpc/examples/routeguide/RouteGuideServer.java#L174) in the example, which is not using two side streaming. Is there any reason to not be able to use just client side streaming? Also you can check the examples I linked, the `routeChat()` one is for two-side streaming.

**Comment by dapengzhang0** [comment]
>We have a gRPC client and server that both are streaming to each other. 

@AviAlche Can you double check the type of your RPC? Is it bidi-streaming or unary or client side streaming? Because the error message

```
 io.grpc.internal.ServerCallImpl internalCloseWARNING: Cancelling the stream with status Status{code=INTERNAL, description=Completed without a response, cause=null}
```

is only for unary and client side streaming. It is never for bidi-streaming RPC.

>Ive got the following lines in the server onComplete one exactly after the other:
responseObserver.onNext(response);
responseObserver.onCompleted();

 Can you double check whether `responseObserver.onNext(response)` is ever executed before the error
occurred? I suspect this is not executed yet at that time. Did you use any server interceptor?

**Comment by creamsoup** [comment]
closing due to inactivity, please feel free to reopen.
